tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4281,161 +4281,140 @@ TEST_F(ValidationTest, AtomicsInvalidDests) {
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15"}, // original: node shader
+      {"!{i32 8, i32 5"},  // changed to: compute shader
+      "Node 'node01' with input/outputs is not compatible with compute", false);
   RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node08 with input/outputs is not compatible with Compute");
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15"}, // original: node shader
+      {"!{i32 8, i32 5"},  // changed to: compute shader
+      "Node 'node02' with input/outputs is not compatible with compute", false);
   RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node09 with input/outputs is not compatible with Compute");
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15"}, // original: node shader
+      {"!{i32 8, i32 5"},  // changed to: compute shader
+      "Node 'node03' with input/outputs is not compatible with compute", false);
   RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node10 with input/outputs is not compatible with Compute");
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15"}, // original: node shader
+      {"!{i32 8, i32 5"},  // changed to: compute shader
+      "Node 'node04' coalescing launch type is not compatible with compute",
+      false);
   RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node11 with input/outputs is not compatible with Compute");
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15, i32 13, i32 2"}, // original: node shader, coalesing
+                                          // launch type
+      {"!{i32 8, i32 5, i32 13, i32 3"},  // changed to: compute shader, thread
+                                          // launch type
+      "Node 'node04' thread launch type is not compatible with compute", false);
```

Comment:
Same here: I think the error should be different.

Reply:
Also, a test with broadcasting launch should be thrown in for good measure, since it shouldn't be defined on a compute shader either, right?